File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        if len(s) != len(t):
            return False

        count = [0] * 26
        for ch1, ch2 in zip(s, t):
            count[ord(ch1)-ord('a')] += 1
            count[ord(ch2)-ord('a')] -= 1

        for num in count:
            if num != 0:
                return False
        return True
        

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        if(s.length() != t.length())
        {
            return false;
        }

        int[] charArr = new int[26];

        for(int i = 0; i<s.length(); i++)
        {
            char ch = s.charAt(i);
            charArr[ch-'a']++;
        }

        for(int j = 0; j<t.length(); j++)
        {
            char ch = t.charAt(j);
            charArr[ch-'a']--;
        }

        for(int ch: charArr)
        {
            if(ch!=0)
                return false;
        }
        return true;
    }
}